packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -906,8 +909,14 @@ class TerminalHandler {
 
   void registerSignalHandlers() {
     assert(residentRunner.stayResident);
-    io.ProcessSignal.SIGINT.watch().listen(_cleanUpAndExit);
-    io.ProcessSignal.SIGTERM.watch().listen(_cleanUpAndExit);
+    io.ProcessSignal.SIGINT.watch().listen((io.ProcessSignal signal) {
+      _cleanUp(signal);
+      io.exit(0);
```

Comment:
Why doesn't throwToolExit work here?

Reply:
That implies that something went wrong, whereas really this is an exit as normal. I'm also not certain that we have integration tests for these process signals so I'd like to keep it as close to what it was before.